submissions/ik-web/js-dom/js/app.js

```diff
@@ -0,0 +1,79 @@
+        },
+        {
+            name: 'Skoda',
+            img: './img/skoda.png',
+            text: `Škoda Auto a.s. (Czech pronunciation: [ˈʃkoda]), commonly called Škoda, is a Czech automobile manufacturer founded in 1895 as Laurin & Klement and headquartered in Mladá Boleslav, Czech Republic. <br> In 1925 Laurin & Klement was acquired by the industrial conglomerate Škoda Works, which itself became state owned in 1948. After 1991 it has been gradually privatized to the German Volkswagen Group, in 1994 became its subsidiary and in 2000 a wholly owned subsidiary. <br> Skoda automobiles are sold in over 100 countries and in 2018, total global sales reached 1.25 million units, an increase of 4.4% from the previous year. The operating profit was €1.6 billion in 2017, an increase of 34.6% over the previous year. As of 2017, Škoda's profit margin was the second highest of all VW Group brands after Porsche.`
+        },
+        {
+            name: 'Mazda',
+            img: './img/mazda.png',
+            text: `Mazda Motor Corporation (commonly referred to as simply Mazda) is a Japanese multinational automaker based in Fuchū, Aki District, Hiroshima Prefecture, Japan. <br> In 2015, Mazda produced 1.5 million vehicles for global sales, the majority of which (nearly 1 million) were produced in the company's Japanese plants, with the remainder coming from a variety of other plants worldwide. In 2015, Mazda was the fifteenth largest automaker by production worldwide.`
+        }
+    ];
+
+    const mainMenu = document.querySelector('.main-menu'); 
+    const cardTitle = document.querySelector('.card-title');
+    const cardImg = document.querySelector('.card-img');
+    const cardText = document.querySelector('.card-text');
+
+    //Menu burger
+    document.querySelector('.burger').addEventListener('click', function(e) {
+        e.preventDefault();
+        this.classList.toggle('active');
+        document.querySelector('.main-menu').classList.toggle('active');
+    });
+
+    //Create a button in main menu
+    let addButtonToMenu = function(item) {
+        const li = document.createElement('li');
+        const button = document.createElement('button');
+
+        //Add the button to main menu
+        mainMenu.appendChild(li.appendChild(button)); 
+
+        // Name and style for button
+        button.classList.add('menu-btn');
+        button.textContent = cars[item].name;
+
+        return button;
+    };
+
+    //Add buttons with event listener to main menu 
+    for (let i = 0; i < cars.length; i++) {
+        addButtonToMenu(i).addEventListener('click', function(e){
```

Comment:
If i click on same button multiple times content will not change but browser will repaint elements.

Reply:
A bit difficult ... probably need to revise the theory again ...

Reply:
You can added some active indicator (class or data atr)
and then check, if active class exists do not execute function which causes rerender

Reply:
I've fixed here but I'm not sure that right... ((